src/components/periodic-table-page/PeriodicTablePage.tsx

```diff
@@ -10,102 +8,90 @@ import PeriodicTable, {
   IPeriodicTableElement,
 } from "../periodic-table/PeriodicTable";
 import PtElementInfo from "../pt-element/PtElementInfo";
-import Modal from "../shared/modal/Modal";
 import Navbar from "../shared/navbar/Navbar";
 import SwipeableModal from "../shared/swipeable-modal/SwipeableModal";
 import ElementInfo from "./element-info/ElementInfo";
 import "./PeriodicTablePage.scss";
 
-type Props = RouteComponentProps<any> & React.Props<any>;
-
 interface IPeriodicTablePageState {
   elementInfo: {
     element: IElement;
     open: boolean;
   };
 }
 
-@autobind
-class PeriodicTablePage extends React.Component<
-  Props,
-  IPeriodicTablePageState
-> {
-  public state: IPeriodicTablePageState = {
+function PeriodicTablePage() {
+  const history = useHistory();
+
+  const [state, setState] = React.useState<IPeriodicTablePageState>(() => ({
     elementInfo: {
       element: ElementManager.getElement(1) as IElement,
       open: false,
     },
-  };
-
-  private ptElements: Map<number, PtElementInfo> = new Map();
+  }));
 
-  public render() {
-    const { elementInfo } = this.state;
-
-    return (
-      <div className="periodic-table-page">
-        <Navbar
-          title={i18n("periodic_table")}
-          className="periodic-table-page__navbar"
-          backButton={true}
-          onBackButtonClick={this.onNavbarBackButtonClick}
-        />
+  const openElementInfo = (element: IElement) => {
+    setState((state) => ({
+      ...state,
+      elementInfo: {
+        element,
+        open: true,
+      },
+    }));
+  };
 
-        <div className="periodic-table-page__table">
-          <PeriodicTable elementRenderer={this.elementRenderer} />
-        </div>
+  const closeElementInfo = () => {
+    const { elementInfo } = state;
+    setState((state) => ({
+      ...state,
+      elementInfo: {
+        ...elementInfo,
+        open: false,
+      },
+    }));
+  };
 
-        <SwipeableModal
-          className="periodic-table-page__modal-element-info"
-          open={elementInfo.open}
-          onClose={this.closeElementInfo}
-        >
-          <ElementInfo element={elementInfo.element} />
-        </SwipeableModal>
-      </div>
-    );
-  }
+  const onNavbarBackButtonClick = () => {
+    history.push(MAIN_MENU);
+  };
 
-  private elementRenderer(atomic: number): IPeriodicTableElement {
+  const elementRenderer = (atomic: number): IPeriodicTableElement => {
     return {
       // @ts-ignore fix this
       component: PtElementInfo,
       props: {
         element: ElementManager.getElement(atomic),
-        onClick: this.elementOnClick,
+        onClick: (element: IElement) => {
+          openElementInfo(element);
+        },
       },
     };
-  }
```

Comment:
You may want to sneak a peek at what's happening in [this PR (#111)](https://github.com/OpenSourceRaidGuild/atom-pwa/pull/111/files#diff-03058fb889f1b9366517962ee8952715a92ea027c14176bdd61eaed3af066fa2)

Reply:
Okay. I see. This has already been conquered. I shall close my pr then.

Reply:
I don't think it cas, just the elementRenderer interface changing (if the PR is merged)

Reply:
Let that PR merge first..I will pull the changes and then open this PR for review. Thanks for the heads up!